### PR TITLE
Improves jousting and gives it to more weapons

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

I was reading over components and saw one called jousting, I vividly remember it being mentioned every so often but I've never seen it in-game. Turns out the SINGLE case for it is when you are using a spear on a borg. The code itself was also very over the place, making it a little confusing to figure out what it did.

I tried cleaning the file up as much as I could, and since I wanted to see this in-game more often, I made some player-facing changes too:

- You can now joust from any vehicle, not just borgs (Secway, ATV, scooter, Charger holoparasite)
- Added jousting to the broom, pitchfork, captain's sabre, and energy sword while active (ONLY esword, NOT desword).
- Added examine text to indicate this feature exists.

Extra notes:
Esword gains half the damage increase and half the knockdown chance than other ways of jousting
Broom only gets 25% damage increase from jousting, since the broom is already pretty strong I thought it would be better off as something used mostly to knockdown.
Spears have to travel a longer distance than other weapons to get their jousting benefits (since it's supposed to be a ghetto weapon)
Jousting now takes the minimum distance needed into account when handling knockdown chance & damage dealt, so travelling 5 tiles will no longer be 100% chance of knockdown if you need a minimum distance of 3 tiles to joust (it will instead be 40%, since you've only traveled 2 tiles in 'jousting' mode).

## Why It's Good For The Game

This is an underused component and I thought it would bring some pretty cool interactions, especially for Holoparasite & Janitors, as a new way to use vehicles to your advantage when it's otherwise seen as just a slowdown.

## Changelog

:cl:
balance: Jousting now works on anything you're buckled to, not just Cyborgs.
balance: Brooms, Pitchforks, the Captain's Sabre, and Energy swords can now be used for jousting.
balance: Spears need to travel a longer distance to joust now.
balance: Jousting's knockdown and damage now only gets stronger after you've traveled the minimum tiles needed to joust.
/:cl:

### CODE REVIEW

**Comment by Rex9001** [comment]
At long last, atv jousting tournaments.

**Comment by Sealed101** [comment]
someone ping lukas, pig jousting must be made real

**Comment by ATH1909** [comment]
note that riding a borg takes up one of your hand slots, so the numbers here should probably be balanced around the one-handed force of these weapons, not the two-handed force.

File: code/datums/components/jousting.dm [hunk 1]
```diff
@@ -1,27 +1,77 @@
...
+
+/datum/component/jousting/proc/on_transform(obj/item/source, mob/user, active)
+	SIGNAL_HANDLER
+	if(!user)
+		return
+
+	if(active)
+		INVOKE_ASYNC(src, PROC_REF(on_equip), user)
+	else
+		INVOKE_ASYNC(src, PROC_REF(on_equip), user)
```
Comment (Jacquerel) [inline code/datums/components/jousting.dm:72]: These are both doing the same thing? Is one supposed to be `on_drop`?

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/datums/components/jousting.dm (hunk 1 above)
Comment (JohnFulpWillard) [reply code/datums/components/jousting.dm:72]: ...yes

**Pull request merged**